vault/external_plugin_test.go

```diff
@@ -277,6 +277,40 @@ func TestCore_EnableExternalPlugin_MultipleVersions(t *testing.T) {
 	}
 }
 
+func TestCore_EnableExternalPlugin_PendingRemoval(t *testing.T) {
+	pluginDir, cleanup := MakeTestPluginDir(t)
+	t.Cleanup(func() { cleanup(t) })
+
+	// create an external plugin to shadow the builtin "app-id"
+	plugin := compilePlugin(t, consts.PluginTypeCredential, "v1.2.3", pluginDir)
+	err := os.Link(path.Join(pluginDir, plugin.fileName), path.Join(pluginDir, "app-id"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	pluginName := "app-id"
```

Comment:
Should we leave a comment here stating that app-id is a deprecated plugin? It may not be obvious to those reading the test code.

Reply:
Alternatively, we could maybe change the name in the mock registry to something like "test-plugin-deprecated"?

Reply:
+1 good idea!